### PR TITLE
Warn about the correct usage of `discrete_space`

### PR DESCRIPTION
**Pull request by tpadioleau**

### CODE REVIEW

File: include/ddc/discrete_space.hpp [hunk 1]
```diff
@@ -194,6 +194,12 @@ std::enable_if_t<2 <= sizeof...(Args), std::tuple<Args...>> init_discrete_space(
     return detail::extract_after(std::move(a), std::index_sequence_for<Args...>());
 }
 
+/**
+ * @tparam DDim a discrete dimension
+ * @return the discrete space instance associated with `DDim`.
+ * It is recommended to call this function from a `KOKKOS_FUNCTION`.
+ * Prefer `ddc::host_discrete_space` for a host-only function.
```
Comment (yasahi-hpc) [inline include/ddc/discrete_space.hpp:201]: This function must be called from a `KOKKOS_FUNCTION`.
Call `ddc::host_discrete_space` for a host-only function instead.

**Review by yasahi-hpc — CHANGES_REQUESTED** [review]
If calling this function inside a host function may cause a problem, it would be better to explicitly claim that.

File: include/ddc/discrete_space.hpp (hunk 1 above)
Comment (tpadioleau) [reply include/ddc/discrete_space.hpp:201]: Yes you are right. The case we encountered is not easy to explain in one sentence. Typically if one wants to retrieved variables from a discrete dimension before a kernel, that is meant to be captured by the lambda, the user "must" call `host_discrete_space`.

**Review by yasahi-hpc — APPROVED** [review]

**Comment by yasahi-hpc** [comment]
Resolves #287

**Pull request merged**